Dockerfile draft

yookoala opened the pull request:
Please merge this repository AFTER you merged #1 

(Or if there is problem, I'll rebase the branch and fix thing before merge)

yookoala committed:
Basic Dockerfile for build

1. Added Dockerfile draft for building the site.
2. Added `scripts/run.sh` to simplify yarn / gulp command calls for Dockerfile..
3. Added documentation for building with docker and locally..
4. Added Travis badge to README.md.

tonyhhyip commented on Dockerfile:
```diff
@@ -0,0 +1,41 @@
+# Version v0.1.0-dev
+
+FROM ubuntu:16.04
```
How about using [node:6](https://hub.docker.com/_/node/) ?

yookoala commented:
Using nvm might allow us to inject node version through env in the future
(without the need to modify Dockerfile) .

2016年12月7日 21:28，"Tony Yip" <notifications@github.com>寫道：

> *@tonyhhyip* commented on this pull request.
> ------------------------------
>
> In Dockerfile
> <https://github.com/tonyhhyip/website-builder/pull/2#pullrequestreview-11799377>
> :
>
> > @@ -0,0 +1,41 @@
> +# Version v0.1.0-dev
> +
> +FROM ubuntu:16.04
>
> How about using node:6 <https://hub.docker.com/_/node/> ?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tonyhhyip/website-builder/pull/2#pullrequestreview-11799377>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAFkiijju2hgKebLjuZuOUw1oRqnM_p4ks5rFrSPgaJpZM4LGc9e>
> .
>

yookoala committed:
gulpfile.js: Fix incorrect webpack error handling

There is no webpack.PluginError. It should be a typo from example
in Webpack's [Usage with Gulp](https://webpack.github.io/docs/usage-with-gulp.html)
documentation page. Used gulp-util accordingly.

yookoala committed:
Added Travis config

Added basic Travis build config to test running docker build and site
build routines. Not artifact verifications yet.

yookoala committed:
Added .dockerignore

yookoala commented:
Added alternative node version support. Also added documentation to README.md

yookoala commented:
Updated Dockerfile to use node:6 as based image now.

yookoala commented:
Also optimised based on [Best Practice documentation](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/)

yookoala commented:
Added multi node-version test on Travis. (v0.10 for docker, also v6 and v7)

tonyhhyip commented:
Merged into the branch docker
